src/browser-realm.ts

```diff
@@ -165,76 +219,181 @@ const IFRAME_ALLOW_ATTRIBUTE_VALUE =
     "xr-spatial-tracking 'none';"
 
 const IFRAME_SANDBOX_ATTRIBUTE_VALUE = 'allow-same-origin allow-scripts';
+const appendChildCall = unapply(Node.prototype.appendChild);
+const removeCall = unapply(Element.prototype.remove);
+const isConnectedGetterCall = unapply((ReflectGetOwnPropertyDescriptor(Node.prototype, 'isConnected') as any).get);
+const { body } = document;
+const createElementCall = unapply(document.createElement);
 
-export default function createSecureEnvironment(distortionMap?: Map<RedProxyTarget, RedProxyTarget>, endowments?: object): (sourceText: string) => void {
+function createDetachableIframe(): HTMLIFrameElement {
     // @ts-ignore document global ref - in browsers
-    const iframe = document.createElement('iframe');
+    const iframe = createElementCall(document, 'iframe');
     iframe.setAttribute('allow', IFRAME_ALLOW_ATTRIBUTE_VALUE);
     iframe.setAttribute('sandbox', IFRAME_SANDBOX_ATTRIBUTE_VALUE);
     iframe.style.display = 'none';
+    appendChildCall(body, iframe);
```

Comment:
here we can introduce the fix for when body is not available and Locker is the first thing loaded in the header of the page:
```
const container = document.childNodes[document.childNodes.length - 1];
appendChild(container, iframe);
```

Reply:
will work on this.

Reply:
Nice!